Tutorial/app/src/main/java/ably/io/tutorial/ExampleActivity.java

```diff
@@ -21,6 +35,68 @@ protected void onCreate(Bundle savedInstanceState) {
         } catch (AblyException e) {
             e.printStackTrace();
         }
+        etUsername = (EditText) findViewById(R.id.editText);
+        btLogin = (Button) findViewById(R.id.button);
+        btLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                new Thread(new Runnable() {
+                    @Override
+                    public void run() {
+                        /* change UI depending on the state */
+                        if (isUserLoggedIn) {
+                            logout();
+                        } else {
+                            login();
+                        }
+                    }
+                }).start();
+            }
+        });
+    }
+
+    private void login() {
+        RequestQueue queue = Volley.newRequestQueue(this);
+        /* 10.0.2.2 is an IP address of the localhost server running on your computer from device emulator perspective */
+        String url = "http://10.0.2.2:3000/login?username=" + etUsername.getEditableText().toString();
+
+        StringRequest stringRequest = new StringRequest(Request.Method.GET, url,
+                new Response.Listener<String>() {
+                    @Override
+                    public void onResponse(String response) {
+                        /* reload activity */
+                        finish();
+                        startActivity(getIntent());
+                    }
+                }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                Toast.makeText(getBaseContext(), "Error, please check if your server is working.", Toast.LENGTH_SHORT).show();
+            }
+        });
+        queue.add(stringRequest);
+    }
+
+    private void logout() {
+        RequestQueue queue = Volley.newRequestQueue(this);
+        /* 10.0.2.2 is an IP address of the localhost server running on your computer from device emulator perspective */
+        String url = "http://10.0.2.2:3000/logout";
```

Comment:
As above, logging out in an app is typically changing the state of the app, not the server.